packages/secure_storage/amplify_secure_storage_dart/lib/src/utils/file_key_value_store.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:file/file.dart' as pkg_file;
+import 'package:file/local.dart' as local_file;
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+
+/// {@template amplify_secure_storage_dart.file_key_value_store}
+/// A Key-Value Store that stores data in the provided file.
+/// {@endtemplate}
+@internal
+class FileKeyValueStore {
+  /// {@macro amplify_secure_storage_dart.file_key_value_store}
+  FileKeyValueStore({required this.directory, required this.fileName});
+
+  /// The directory of the file.
+  final String directory;
+
+  /// The name of the file.
+  ///
+  /// The file will be created if it does not yet exist.
+  final String fileName;
+
+  late final File file = fs.file(
```

Comment:
If `accessGroup` is set, this file could potentially be written to from multiple processes, right? In that case, I don't think this would work. Maybe accessGroup becomes the name of a folder and each appId gets an individual file. If `accessGroup` is a Linux group name, then the folder's permissions could be configured around that.

Reply:
Oh nvm this is only used for the one flag. The appID is still used for everything else - my b

Reply:
I think you bring up a good point. We don't want to clear keys if an access group is set. With the current impl, if two apps use the same access group, the second app will clear all the keys set by the first app.

I will update this to skip initialization if there is an access group set. I think that is all we can do since App N will have no way to know if another app set those keys, or if they were from a previous install.

Reply:
I have updated this in the linux impl